src/store/global/global.thunks.js

```diff
@@ -240,6 +247,37 @@ function checkPendingDelayedWithdraw (exitId) {
   }
 }
 
+/**
+ * Checks pending exits to see if they have been completed
+ * and delete them from storage
+ * @returns {void}
+ */
+function checkPendingWithdrawals () {
+  return (dispatch, getState) => {
+    const { global: { wallet, ethereumNetworkTask, pendingWithdraws } } = getState()
+
+    dispatch(globalActions.checkPendingWithdrawals())
+
+    const pendingWithdrawsAccount = storage.getItemsByHermezAddress(
+      pendingWithdraws,
+      ethereumNetworkTask.data.chainId,
+      wallet.hermezEthereumAddress
+    )
+
+    return Promise.all(pendingWithdrawsAccount.map((pendingWithdraw) => {
+      return CoordinatorAPI.getExit(pendingWithdraw.batchNum, pendingWithdraw.accountIndex)
+        .then((exitTx) => {
+          if (exitTx.instantWithdraw || exitTx.delayedWithdraw) {
+            dispatch(removePendingWithdraw(pendingWithdraw.id))
+          }
+        })
+        .catch(() => {})
+    }))
+      .then(() => dispatch(globalActions.checkPendingWithdrawalsSuccess()))
+      .catch(() => dispatch(globalActions.checkPendingWithdrawalsSuccess()))
```

Comment:
Why success is returned in both cases?

Reply:
It's fine if it fails, this is just an automated check to clean up pending withdrawals. I copied the structure from the funcion `checkPendingDeposits` that Elias did